opt_einsum/path_random.py

```diff
@@ -0,0 +1,320 @@
+import math
+import time
+import heapq
+import random
+import functools
+from collections import deque
+
+from . import helpers
+from .paths import PathOptimizer, _ssa_optimize, _BETTER_FNS, ssa_to_linear, _calc_k12_flops
```

Comment:
Usually we try not to import `_*` functions. Maybe convert add a getter function to `_BETTER_FNS` and move those others to non `_`?